compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -839,29 +858,10 @@ impl<'hir> Map<'hir> {
     }
 
     pub fn opt_span(self, hir_id: HirId) -> Option<Span> {
-        fn until_within(outer: Span, end: Span) -> Span {
-            if let Some(end) = end.find_ancestor_inside(outer) {
-                outer.with_hi(end.hi())
-            } else {
-                outer
-            }
-        }
-
-        fn named_span(item_span: Span, ident: Ident, generics: Option<&Generics<'_>>) -> Span {
-            if ident.name != kw::Empty {
-                let mut span = until_within(item_span, ident.span);
-                if let Some(g) = generics
-                    && !g.span.is_dummy()
-                    && let Some(g_span) = g.span.find_ancestor_inside(item_span)
-                {
-                    span = span.to(g_span);
-                }
-                span
-            } else {
-                item_span
-            }
+        if let Some(owner) = hir_id.as_owner() {
+            let span = self.tcx.def_span(owner.def_id);
+            return Some(span);
         }
-
         let span = match self.find(hir_id)? {
```

Comment:
Should we remove all the HIR owners from this match? At first sight, only the fallback `span_with_body` case should remain.

Reply:
I'm uncertain whether it's appropriate, for instance:

```rs
// issue-47706.rs
enum Qux {
    Bar(i32),
}

fn foo<F>(f: F)
where
    F: Fn(),
{}

fn main() {
    foo(Qux::Bar);
}
```

When we aim to obtain the span of ﻿`Bar`, which is of type `﻿Ctor`, we require this logic.

Reply:
Makes sense, Ctor is not a HIR owner. Could you still remove HIR owners?
Other possibility: make the fast path use `tcx.local_def_id` instead of `as_owner`, and remove Ctor too.
I don't have a strong opinion on that possibility.

Reply:
> Could you still remove HIR owners?

I'm a bit confused. Are you suggesting that we delete this branch?

```rs
if let Some(owner) = hir_id.as_owner() {
    let span = self.tcx.def_span(owner.def_id);
    return Some(span);
}
```


> use `tcx.local_def_id`

`﻿local_def_is` is not an associated method of `﻿tcx`. It belongs to `﻿LoweringContext`..

Reply:
> I'm a bit confused. Are you suggesting that we delete this branch?

The opposite. I'm suggesting to delete the code that not reachable any more thanks to this branch.

> ﻿local_def_is is not an associated method of ﻿tcx. It belongs to ﻿LoweringContext..

My bad. I had not seen it had been removed from `tcx.hir()`.